modules/service/src/main/scala/lucuma/odb/service/CalibrationsService.scala

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) 2016-2023 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.odb.service
+
+import cats.FlatMap
+import cats.syntax.all.*
+import lucuma.core.enums.GmosNorthFpu
+import lucuma.core.enums.GmosNorthGrating
+import lucuma.core.enums.GmosSouthFpu
+import lucuma.core.enums.GmosSouthGrating
+import lucuma.core.enums.GmosXBinning
+import lucuma.core.enums.GmosYBinning
+import lucuma.core.math.Wavelength
+import lucuma.core.model.Program
+import lucuma.odb.service.Services.Syntax.*
+import lucuma.odb.util.Codecs.*
+import lucuma.odb.util.GmosCodecs.*
+import org.typelevel.log4cats.Logger
+import skunk.Query
+import skunk.syntax.all.*
+
+trait CalibrationsService[F[_]] {
+  def recalculateCalibrations(
+    pid: Program.Id
+  )(using L: Logger[F]): F[Unit]
+
+}
+
+object CalibrationsService {
+  def instantiate[F[_]: FlatMap](using Services[F]): CalibrationsService[F] =
+    new CalibrationsService[F] {
+      def recalculateCalibrations(pid: Program.Id)(using L: Logger[F]): F[Unit] =
+        for {
+          gnls   <- session.execute(Statements.SelectGmosNorthLongSlitConfigurations)(pid)
```

Comment:
For now we just need the configurations, but we need to add in the future the actual work